### PR TITLE
Embedded variables jwt_header and jwt_grant

### PR DESCRIPTION
**Pull request by RekGRpth**

### CODE REVIEW

File: src/ngx_http_auth_jwt_module.c [hunk 1]
```diff
@@ -504,3 +549,80 @@ static ngx_int_t auth_jwt_get_token(u_char **token, ngx_http_request_t *r, const
...
+    v->not_found = 0;
+    return NGX_OK;
+}
+
+static ngx_int_t ngx_http_auth_jwt_header_var(ngx_http_request_t *r, ngx_http_variable_value_t *v, uintptr_t data) {
+    v->not_found = 1;
+    jwt_t *jwt = ngx_http_get_module_ctx(r, ngx_http_auth_jwt_module);
+    if (!jwt) return NGX_OK;
+    ngx_str_t *name = (ngx_str_t *)data;
+    size_t len = name->len - (sizeof("jwt_header_") - 1);
```
[the remark this answers is not in the document]
Comment (RekGRpth) [reply src/ngx_http_auth_jwt_module.c:595]: "jwt_header_abcd".length = 15
sizeof("jwt_header_") = 12
15 - (12 - 1) = 4

[the remark this answers is not in the document]
Comment (max-lt) [reply src/ngx_http_auth_jwt_module.c:595]: Right

File: src/ngx_http_auth_jwt_module.c [hunk 2]
```diff
@@ -504,3 +549,80 @@ static ngx_int_t auth_jwt_get_token(u_char **token, ngx_http_request_t *r, const
...
+    char header[len + 1];
```
[the remark this answers is not in the document]
Comment (RekGRpth) [reply src/ngx_http_auth_jwt_module.c:596]: no, but 5!

**Comment by pjz** [comment]
Any idea when/if this will be merged? I'd like to use it, and a merge would avoid me setting up my own fork and etc.

**Comment by RekGRpth** [comment]
> Any idea when/if this will be merged? I'd like to use it, and a merge would avoid me setting up my own fork and etc.

Try use https://github.com/maxam18/nginx-ejwt-module instead

**Comment by max-lt** [comment]
It won't be, as this PR came with no description, no example case, no tests, no comments.

The added code does't fit the project style and is a bit obscure; 2 of 4 functions are copy-pasted of the 2 others and I think it can be done much more cleanly. I tried to clear this up at the time but finaly gave up.

Furthermore, I don't see any interest in getting JWT variables in the Nginx context (except for logging, maybe ?). I think that making security operations in this scope (if that's your idea) may be tricky (especially with the few tools Nginx provides, remember : [if is evil](https://www.nginx.com/resources/wiki/start/topics/depth/ifisevil/)).

Feel free to fork the project from this point and improve it.

**Pull request closed without merging**